### ISSUE TITLE
[Feature Request]: Firestick/ Gamepad navigation support

### ISSUE THREAD

**Issue by Vintagemotors** [MarmadileManteater/FreeTubeCordova#283]
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

When attempting to navigate with a standard Firestick remote, the directional pad acts as though it is tabbing through the interface rather than using a cursor which results in not being able to scroll properly through sections of the settings menu and having difficulty using player controls. 

### Proposed Solution

Implement a virtual cursor as is used in FireTV/ AndroidTV native web browsers. 

### Alternatives Considered

I tried using Mouse Toggle which sort of works for some Android apps but in this case it was unable to cleanly scroll or interact with player controls. There are other similar apps but none work anywhere near as well as native support. 

### Issue Labels

ease of use improvement, new feature, new optional setting

### Additional Information

This is the source code for the Mozilla Firefox TV edition which has this functionality and may provide some implementation reference: https://github.com/mozilla-mobile/firefox-tv It Is currently being worked on upstream for SteamDeck but it is unclear as to whether that implementation would work on AndroidTV after being merged. https://github.com/FreeTubeApp/FreeTube/issues/4277

**Comment by MarmadileManteater** [comment]
https://developer.amazon.com/docs/fire-tv/supporting-controllers-in-web-apps.html 

From what I am reading here, it seems like the remote is bound to keyboard inputs (in webviews). I am unsure as to why this would result in the tabbing behavior. It also supports the native browser apis for controllers, so I'm assuming that upstream PR will help with support of gamepads connected to the fire TV, but not the remote.

I may have to obtain a firestick to easily address these issues, and I don't currently own one. Although, as far as android testing devices go, fire sticks are much less expensive.

**Comment by gripfastistech** [comment]
2nd. Unusable on Hisense Android Google TV with DS4 controller. There is no cursor/tab hover indication so all controlled is invisible if not using trackpad.

**Comment by MarmadileManteater** [comment]
@gripfastistech Does it show cursor indicators in other webviews? (Not like your web browser, but other webview apps)

**Comment by gripfastistech** [comment]
Yes. BetterXC does though that targets controller support by design for use with GamePass X-cloud. It uses chromium as a base I believe (?).

Jul 22, 2024 4:28:46 AM Emma ***@***.***>:

**Comment by Slowdive1123** [comment]
I wanted to echo what some other people have said. This app looks amazing, but is essentially unusable for AndroidTV. Would be wonderful if a few tweaks could be made to make it more user friendly. Thank you.

**Comment by Nico7as** [comment]
I've tried several Youtube apps made for vanilla Android on an Android TV device, and got issues with everyone (NewPipe, LibreTube)

For now I use SmartTube, which is a dedicated Android TV app for Youtube.
It's quite limited in functions (no subscriptions without a Youtube Account), but it works well with a remote.

I'd love to have a FreeTube app for Android TV !

**Comment by MarmadileManteater** [comment]
Okay, I now have my hands on a fire stick, so I have been able to see how the app runs on a tv type device, and my opinion is that fixing navigation is not really enough. The problem fundamentally is that certain system things don't work with the remote, such as the file picker. I can't do a whole lot about that afaik unless I refactor the code to be able to work with the permission to access all files.

**Comment by MarmadileManteater** [comment]
didn't mean to close this issue

**Comment by MarmadileManteater** [comment]
Idk, I might work on this. I might not. Its not very high on my priority list, but I, at least, have a device to test it.

**Comment by MarmadileManteater** [comment]
Its definitely lower on my priority list than getting that hotfix out that I'm still slacking on.

**Comment by MarmadileManteater** [comment]
I am considering my options. I think the app could be a lot closer to working well with a few small tweaks, but there are big blockers as well keeping it from being the best it could be.

**Comment by Nico7as** [comment]
Glad to know that you're looking into it.

I'd be glad to test a tweaked version, and give some feedback.
I'd be using the official remote, not a gamepad.
